.buildkite/x-pack/pipeline.xpack.filebeat.yml

```diff
@@ -18,62 +18,311 @@ env:
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2019}"
+          machineType: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "x-pack/filebeat/build/*.xml"
+          - "x-pack/filebeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "x-pack/filebeat: Windows 2019 Unit Tests"
 
-  - input: "Input Parameters"
-    key: "force-run-stages"
-    fields:
-    - select: "Filebeat - run_xpack_filebeat"
-      key: "run_xpack_filebeat"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    - select: "Filebeat - run_xpack_filebeat_macos_tests"
-      key: "run_xpack_filebeat_macos_tests"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    - select: "Filebeat - run_xpack_filebeat_arm_tests"
-      key: "run_xpack_filebeat_arm_tests"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    - select: "Filebeat - run_xpack_filebeat_aws_tests"
-      key: "run_xpack_filebeat_aws_tests"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-
-    if: "build.source == 'ui'"
+      - label: ":windows: Windows 10 Unit Tests"
+        command: |
+          Set-Location -Path x-pack/filebeat
+          mage build unitTest
+        key: "x-pack-filebeat-extended-win-10-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_10}"
+          machineType: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "x-pack/filebeat/build/*.xml"
+          - "x-pack/filebeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "x-pack/filebeat: Windows 10 Unit Tests"
+
+      - label: ":windows: Windows 11 Unit Tests"
+        command: |
+          Set-Location -Path x-pack/filebeat
+          mage build unitTest
+        key: "x-pack-filebeat-extended-win-11-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_11}"
+          machineType: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "x-pack/filebeat/build/*.xml"
+          - "x-pack/filebeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "x-pack/filebeat: Windows 11 Unit Tests"
+
+  - group: "x-pack/filebeat Extended Tests"
+    key: "x-pack-filebeat-extended-tests"
+    steps:
+      - label: ":mac: MacOS x86_64 Unit Tests"
+        if: build.env("BUILDKITE_PULL_REQUEST") == "false"  || build.env("GITHUB_PR_LABELS") =~ /.*(macOS|aws).*/
```

Comment:
if we enable `if: build.env("BUILDKITE_PULL_REQUEST") == "false"` at the group level we can remove it from here

Reply:
this is how we had it for Filebeat as well.